### ISSUE TITLE
SkiaStringSizeService and measuring strings in general seems off.

### ISSUE THREAD

**Issue by lucapivato** [dotnet/Microsoft.Maui.Graphics#459]
Apparently Skia is used in Chrome as well. However, it seems that the measuring of strings in Microsoft.Maui.Graphics.Skia.SkiaStringSizeService incorrectly uses the passed in font size as the height.

In Chrome or Edge

var el = document.createElement("div");
document.body.append(el);
el.innerText = "Hello, Maui.Graphics!"
el.style.fontFamily="Arial"
el.style.fontSize="13px"
el.clientHeight
>15

15 is the correct height of the text when setting the font size to 13px for Arial. This value is returned by Chrome and Edge, both apparently using Skia. 

I get the same correct value the good old gdi+ and libgdiplus setting the font size to 13 pixels, the height is 15.

![image](https://user-images.githubusercontent.com/37640802/175563033-0ea9cdd0-e398-4e0f-ac67-211ff373fbc2.png)

But Microsoft.Maui.Graphics.Skia.SkiaStringSizeService always returns the incorrect height simply returning the size.

Is there a settings I'm missing or an additional way to measure a string correctly?

**Comment by janseris** [comment]
Use libgdiplus. this library is broken and incomplete

**Comment by lucapivato** [comment]
Yes we are. I’m planning for a feasible replacement. Ms seems determined to
block usage of libgdiplus at some point. Aspose has a fully managed
system.drawing library but the oem license is unworkable for us.

Another option is to simply take over libgdiplus code for our needs (which
are very modest). The c code is relatively small and well written and easy
to maintain for c/c++ developers. Cairo seems to work well too.

Otherwise I’d have to dig into chromium’s usage of skia to see how it
measures text blocks (with horizontal constraints, etc) since it seems not
to be implemented in skia by itself. But I may be wrong since I haven’t
spent time on this part yet.


-- 

Ice Tea Group
http://wisej.com
http://iceteagroup.com
+1-202-449-3778 x701
